### ISSUE TITLE
Allow separate permissions when adding file sets to works

### ISSUE THREAD

**Issue by awead** [samvera-labs/hyrax#1134]
### Descriptive summary

When adding new files to existing works, one cannot specify a visibility or set of permissions for the new file that is different than the work.

When uploading a new file and supplying a different set of permissions/visibility, the new changes are applied to the work, and (optionally) all contained files, or just the new one. This last part isn't clear in the wording of the page when it asks you to if you want to copy permissions.

### Rationale

Users might want to upload a file with a different set of permissions. This is a question.

**Comment by mjgiarlo** [comment]
@awead Is this a theoretical need or does Penn State have a use case for this?

What would you think about raising this for discussion on a Samvera Tech call or on the -tech/-community lists?

**Comment by mjgiarlo** [comment]
Needs design.

**Comment by mjgiarlo** [comment]
Open questions:

* Would the perms controls also appear when editing a work?
* How do we handle perm propagations when a FS has different perms from its containing work?

**Comment by awead** [comment]
@mjgiarlo We don't have a use case, I just noticed it today when I was testing Hyrax out and thought it might need some discussion. The specific case I had in mind was, public work with some public files, but I want to add a private file. I can, but it's a two-step process: 1) add the file as a public file, then, 2) edit it to be private.

It's probably low on the list of priorities. It's also trickier than it seems. What if I want to embargo or lease a file differently from other files and the work? Why? Anyway, just bringing this up here.

**Comment by mjgiarlo** [comment]
@awead Yep! I agree that it's a good one to discuss so we can come up with a coherent and sound technical design. Thanks for raising this.

**Comment by atz** [comment]
"Public first, private later" is a recipe for embarrassment.  Since we are talking about potentially publication and tenure-busting exposure, I would recommend against that path.  

I am totally OK with allowing the batch functionality to stay with one permission setting.  If finer variation is needed, those can be handled via individual actions separate from the bulk.  We don't need **all** the complexity in batch.

**Comment by rjkati** [comment]
This is still present as described in Hyrax 3.0.2. Users cannot add a file which is a different visibility than the work. They can add a file which is the same visibility as the work and then edit the file's visibility via the file set edit page.
